File: Backend/fill_examples/redirects.py
def main(args):
    """
    Example: HTTP Redirects
    Demonstrates how to redirect users to different URLs.
    """
    # Get the redirect target from the request
    target = args.get("body", {}).get("target", "https://example.com")
    redirect_type = args.get("body", {}).get("type", "temporary")
    
    # Determine status code (301 = permanent, 302 = temporary)
    status_code = 301 if redirect_type == "permanent" else 302
    
    # Return redirect response
    return {
        "_shsf": "v2",
        "_code": status_code,
        "_location": target
    }

# Example use cases:
# - Redirecting after login: return {"_code": 302, "_location": "/dashboard", "_shsf": "v2"}
# - URL shortener: return {"_code": 301, "_location": "https://long-url.com/page", "_shsf": "v2"}
# - Form submission redirect: return {"_code": 302, "_location": "/thank-you", "_shsf": "v2"}

File: Backend/fill_examples/redirects.go
package main

// Example: HTTP Redirects
// Demonstrates how to redirect users to different URLs.
func main_user(args interface{}) (interface{}, error) {
	// Type assert to map to access data
	payload, ok := args.(map[string]interface{})
	if !ok {
		return map[string]interface{}{
			"_shsf": "v2",
			"_code": 400,
			"_res": map[string]interface{}{
				"state": false,
				"error": "Invalid payload format",
			},
		}, nil
	}

	// Get the redirect target from the request
	body, _ := payload["body"].(map[string]interface{})
	target := "https://example.com"
	if t, ok := body["target"].(string); ok {
		target = t
	}

	redirectType := "temporary"
	if rt, ok := body["type"].(string); ok {
		redirectType = rt
	}

	// Determine status code (301 = permanent, 302 = temporary)
	statusCode := 302
	if redirectType == "permanent" {
		statusCode = 301
	}

	// Return redirect response
	return map[string]interface{}{
		"_shsf":     "v2",
		"_code":     statusCode,
		"_location": target,
	}, nil
}

// Example use cases:
// - Redirecting after login: return map[string]interface{}{"_code": 302, "_location": "/dashboard", "_shsf": "v2"}
// - URL shortener: return map[string]interface{}{"_code": 301, "_location": "https://long-url.com/page", "_shsf": "v2"}
// - Form submission redirect: return map[string]interface{}{"_code": 302, "_location": "/thank-you", "_shsf": "v2"}
